aptos-move/framework/aptos-framework/sources/Account.move

```diff
@@ -343,18 +343,70 @@ module AptosFramework::Account {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    /// Basic account creation method.
+    /// Basic account creation methods.
     ///////////////////////////////////////////////////////////////////////////
 
     public(script) fun create_account(auth_key: address) {
-        let (signer, _) = create_account_internal(auth_key);
+        let signer = create_account_internal(auth_key);
         Coin::register<TestCoin>(&signer);
     }
 
+    public fun create_module_account(
+        source: &signer,
```

Comment:
Can we consider calling this function as part of deploying a module - basically when someone deploy a module, we create a brand new separate account for that module and then hand the SignerCapability to the account deploying it by default? This is definitely a lot more work but can make things a bit easier.

Reply:
You mean as part of the initialize function? Maybe we could move down that path. Honestly though, I'm not that concerned. These accounts are truly randomly distributed in the system. There's no more additional cost than adding a new entry to a table.

Reply:
The reason I think we should explore this use case for modules more, as I also mentioned in earlier comments, is to allow modules to exist separately from an account with private keys. This allows building decentralized systems more easily.

Reply:
Same as previous comment